### PR TITLE
Errors in the web view don't crash the process without responding.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
When we have a situation that an unknown exception is thrown in this view, we throw it out into the web view, which
has poor characteristics about handling the error.  Instead, if it is not a critical error, catch it and report it to the callback
listener.  If it is a critical Error, report it and rethrow it.

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/exception/ErrorStrings.java [hunk 1]
```diff
@@ -220,6 +220,16 @@ private ErrorStrings() {
      */
     public static final String KEY_CHAIN_PRIVATE_KEY_EXCEPTION = "Key Chain private key exception";
 
+    /**
+     * Key Chain public key exception.
+     */
+    public static final String KEY_CHAIN_PUBLIC_KEY_EXCEPTION = "Key Chain public key exception";
+
```
Comment (AdamBJohnsonx) [inline common4j/src/main/com/microsoft/identity/common/java/exception/ErrorStrings.java:227]:
```suggestion
```

**Comment by AdamBJohnsonx** [comment]
> Do we know why the public key was null?

In the key information refactoring that was done, it was not persisted.  There are no current uses, and it should be part of the persisted certificate.

File: common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallenge.java [hunk 2]
```diff
@@ -179,12 +181,17 @@ public String getSubmitUrl() {
                 if (privateKey == null) {
                     throw new ClientException(ErrorStrings.KEY_CHAIN_PRIVATE_KEY_EXCEPTION);
                 }
+                final PublicKey publicKey = deviceCertProxy.getPublicKey();
+                final X509Certificate certificate = deviceCertProxy.getCertificate();
+                if (certificate == null) {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallenge.java:186]: What if the device isn't joined? Or we are not going to even get here in that case?

**Review by rpdome — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallenge.java (hunk 2 above)
Comment (rpdome) [reply common4j/src/main/com/microsoft/identity/common/java/challengehandlers/PKeyAuthChallenge.java:186]: the cert proxy should not be set in the first place

**Pull request merged**